dj/api/nodes.py

```diff
@@ -44,6 +47,74 @@ class NodeMetadata(SQLModel):
     columns: List[SimpleColumn]
 
 
+class NodeValidation(SQLModel):
+    """
+    A validation of a provided node definition
+    """
+
+    message: str
+    status: NodeStatus
+    node: Node
+    dependencies: List[NodeMetadata]
+    columns: List[Column]
+
+
+@router.post("/nodes/validate/", response_model=NodeValidation)
+def validate_node(
+    data: Union[NodeBase],
+    session: Session = Depends(get_session),
+) -> NodeValidation:
+    """
+    Validate a node.
+    """
+
+    if data.type == NodeType.SOURCE:
+        raise DJException(message="Source nodes cannot be validated")
+    node = Node.from_orm(data)
+
+    # Try to parse the node's query and extract dependencies
+    try:
+        (
+            query_ast,
+            dependencies_map,
+            missing_parents_map,
+        ) = extract_dependencies_from_node(
+            session=session,
+            node=node,
+            raise_=False,
+        )
+    except ValueError as exc:
+        raise DJException(message=str(exc)) from exc
+
+    # Only raise on missing parents if the node mode is set to published
+    if missing_parents_map and node.mode == NodeMode.PUBLISHED:
+        raise DJException(
+            errors=[
+                DJError(
+                    code=ErrorCode.MISSING_PARENT,
+                    message="Node definition contains references to nodes that do not exist",
+                    debug={"missing_parents": list(missing_parents_map.keys())},
+                ),
+            ],
+        )
+
+    # Add aliases for any unnamed columns and confirm that all column types can be inferred
+    query_ast.select.add_aliases_to_unnamed_columns()
```

Comment:
tangential: do we intend to write out the query again which would include these added aliases? I was also experimenting with some serialization just for [reference](https://github.com/CircArgs/dj/blob/8ea622c230437b8142ae8287ed4a6161df2ce11b/dj/sql/parsing/ast.py#L197)

Reply:
Hmm, this is a great point I hadn't thought of. I can see the upside of automatically aliasing unnamed queries and then saving them as already aliased. I'm wondering if there are any drawbacks. Downstream transformations will have to reference the generated column name, i.e. `_col1` but that was always true, right?

Reply:
yeah that was always true. the purpose of actually doing this aliasing in my mind is to make sure that there is consistency when a query is run regardless of the database.